haystack/components/generators/chat/hugging_face_api.py

```diff
@@ -159,6 +163,11 @@ def __init__(  # pylint: disable=too-many-positional-arguments
             msg = f"Unknown api_type {api_type}"
             raise ValueError(msg)
 
+        if tools:
+            if streaming_callback is not None:
```

Comment:
Should we do this across the board for all CG - just something to write down and not forget

Reply:
No. Most of Generators support Tools + Streaming, I think.

Theoretically also HF could support this but it is highly undocumented and does not behave consistently.
See https://github.com/deepset-ai/haystack-experimental/pull/120#discussion_r1806334949